### PR TITLE
Hrcpp 390/rm some expected fail

### PR DESCRIPTION
**Pull request by rigazilla**
This PR contains changes on the swig wrapper that remove some of expected failure in the java test list.
Plus a fix on the client that now correctly set the force return value flag in the request.

https://issues.jboss.org/browse/HRCPP-390

### CODE REVIEW

**Comment by rigazilla** [comment]
I've removed the tabs.

It seems that you had an infinispan server already running before you run the test suite. This conflicts with the TLS and SASL startup.

**Comment by rigazilla** [comment]
adding @mgencur as reviewer since he's the reporter of this issue.

**Comment by alanfx** [comment]
@rigazilla ok, code and tests look good to me now. 👍

**Comment by rigazilla** [comment]
@alanfx if it's all ok can you merge? thanx!

File: jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java [hunk 1]
```diff
@@ -158,18 +155,17 @@ when main() returns. */
 
 @Override
 public boolean includeMethod(IMethodSelectorContext context, ITestNGMethod method, boolean isTestMethod) {
```
Comment (alanfx) [inline jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java]: It looks like the indentation for `includeMethod` and `setTestMethods` is off?

File: jni/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java [hunk 2]
```diff
@@ -178,9 +178,8 @@ public ConfigurationBuilder marshaller(Class<? extends Marshaller> marshaller) {
```
Comment (alanfx) [inline jni/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java]: Should this^^^ version of the method work too?

Comment (rigazilla) [reply jni/src/main/java/org/infinispan/client/hotrod/configuration/ConfigurationBuilder.java]: ok enabled

File: jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java (hunk 1 above)
Comment (rigazilla) [reply jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java]: ok, fixed

File: jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java [hunk 3]
```diff
@@ -17,19 +17,19 @@
 import org.testng.reporters.TextReporter;
```
[the remark this answers is not in the document]
Comment (rigazilla) [reply jni/src/test/java/org/infinispan/client/jni/hotrod/JniTest.java]: ok, fixed

**Comment by alanfx** [comment]
Looks good to me. Merging...

**Pull request merged**